flink-core/src/main/java/org/apache/flink/api/common/state/ListStateDescriptor.java

```diff
@@ -84,21 +138,37 @@ public boolean equals(Object o) {
 
 		ListStateDescriptor<?> that = (ListStateDescriptor<?>) o;
 
-		return serializer.equals(that.serializer) && name.equals(that.name);
-
+		return elemTypeSerializer.equals(that.elemTypeSerializer) && name.equals(that.name);
 	}
 
 	@Override
 	public int hashCode() {
-		int result = serializer.hashCode();
+		int result = elemTypeSerializer.hashCode();
 		result = 31 * result + name.hashCode();
 		return result;
 	}
 
 	@Override
 	public String toString() {
 		return "ListStateDescriptor{" +
-				"serializer=" + serializer +
+				"elem serializer=" + elemTypeSerializer +
 				'}';
 	}
+
+	// ------------------------------------------------------------------------
+	//  Serialization
+	// ------------------------------------------------------------------------
+
+	private void writeObject(final ObjectOutputStream out) throws IOException {
+		// make sure we have a serializer before the type information gets lost
+		initializeSerializerUnlessSet(new ExecutionConfig());
+
+		// write all the non-transient fields
+		out.defaultWriteObject();
+	}
+
+	private void readObject(final ObjectInputStream in) throws IOException, ClassNotFoundException {
```

Comment:
I think this method is not necessary, because it only triggers default serialization anyways.